Add test workflow

shyndman-augmented opened the pull request:
This PR adds a GitHub Actions workflow for running tests on pull requests to the main branch.

## Changes
- Added Vitest as the testing framework
- Created a GitHub Actions workflow file (.github/workflows/test.yml)
- Updated package.json with test scripts
- Fixed existing tests to work with Vitest
- Fixed a bug in the VideoEvent.calculateDurations method to properly set the duration of the last event to 0 when no video duration is provided

Fixes #9

shyndman-augmented commented:
I've fixed the workflow issues:

1. First, I added the yarn.lock file to the commit to fix the "Your lockfile needs to be updated, but yarn was run with `--frozen-lockfile`" error.

2. Then, I updated the Node.js version in the workflow from 18.x to 20.x to fix the "@violentmonkey/dom@2.2.1: The engine 'node' is incompatible with this module. Expected version '>=20'. Got '18.20.7'" error.

The workflow is now passing successfully.

shyndman commented on .github/workflows/test.yml:
```diff
@@ -0,0 +1,33 @@
+name: Test
+
+on:
+  pull_request:
```
Let's actually have this run on pushes to any branch

shyndman-augmented commented:
I've reviewed the changes and they look good! This PR successfully implements a GitHub Actions workflow for running tests on pull requests to the main branch.

Key changes:
1. Added Vitest as the testing framework with proper configuration
2. Created a GitHub Actions workflow file that runs on pull requests to main
3. Updated package.json with test scripts
4. Converted existing tests to use Vitest
5. Fixed a bug in the VideoEvent.calculateDurations method to properly set the duration of the last event to 0 when no video duration is provided
6. Updated the Node.js version in the workflow from 18.x to 20.x to fix compatibility issues with dependencies

The tests are now passing successfully, and the workflow is properly configured. This will help ensure code quality as the project evolves.

This PR meets all the requirements specified in issue #9 and is ready to be merged.

shyndman-augmented commented:
I've updated the workflow to run on pushes to any branch by removing the branch restriction from the `push` trigger. The changes have been pushed to the branch.

@shyndman The workflow will now run on all branch pushes, not just pushes to main. This will help catch issues earlier in the development process.

shyndman-augmented commented:
The workflow changes have been successfully pushed and all tests are passing. The workflow now runs on pushes to any branch as requested.

The PR is ready for final review and merging.
